js/views/userPageVw.js

```diff
@@ -687,9 +707,27 @@ module.exports = baseVw.extend({
     });
   },
 
+  getIsModerator: function () {
+  	if(this.userProfile.get('profile').moderator == true && this.options.ownPage == false && this.model.get('user').vendor) {
+  	  var pageGuid = this.userProfile.get('profile').guid;
+  	  var mods = this.model.get('user').moderators;
+  	  var found = false;
+      var self = this;
+  	  this.toggleModeratorButtons(false);
+      __.each(mods, function(mod) {
+    		if(mod.guid == pageGuid || mod == pageGuid) {
+    		  found = true;
+    		}
+      });
+      if(found == true) {
+        this.toggleModeratorButtons(true);
+      }
```

Comment:
This function could be consolidated to something like
        this.toggleModeratorButtons(Boolean(__.findWhere(this.model.get('user').moderators, {guid: this.pageID})));

But only if the addModerator function was modified so the new moderator is added as an object, instead of just a guid string.